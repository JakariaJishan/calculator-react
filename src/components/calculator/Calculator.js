/*eslint-disable */
import React, { useState } from "react";
import calculate from "../../logic/calculate";
import "./calculator.css";

const Calculator = () => {
  const [obj, setObj] = useState({
    total: 0,
  });
  const clickHandler = (e) => {
    const name = e.target.getAttribute("name");
    setObj(calculate(obj, name));
  };

  const { total, next, operation } = obj;
  return (
    <div className="calculator">
      <h2 className="calc-heading">Lets do some Math...</h2>
      <div className="wrapper">
        <div className="output">
          {next || operation || total ? (
            <p>
              {total} {operation} {next}
            </p>
          ) : (
            0
          )}
        </div>
        <div className="keys">
          <button
            className="ac"
            name="AC"
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            AC
          </button>
          <button
            className="plus-minus"
            name="+/-"
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            +/-
          </button>
          <button
            className="percent"
            name="%"
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            %
          </button>
          <button
            className="division"
            name="รท"
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            รท
          </button>
          <button
            className="seven"
            name="7"
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            7
          </button>
          <button
            className="eight"
            name="8"
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            8
          </button>
          <button
            className="nine"
            name="9"
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            9
          </button>
          <button
            className="multiply"
            name="x"
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            x
          </button>
          <button
            className="four"
            name="4"
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            4
          </button>
          <button
            className="five"
            name="5"
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            5
          </button>
          <button
            className="six"
            name="6"
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            6
          </button>
          <button
            className="minus"
            name="-"
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            -
          </button>
          <button
            className="one"
            name="1"
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            1
          </button>
          <button
            className="two"
            name="2"
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            2
          </button>
          <button
            className="three"
            name="3"
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            3
          </button>
          <button
            className="plus"
            name="+"
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            +
          </button>
          <button
            className="zero"
            name="0"
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            0
          </button>
          <button
            className="dot"
            name="."
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            .
          </button>
          <button
            className="equal"
            name="="
            onClick={clickHandler}
            onKeyDown={clickHandler}
            aria-hidden="true"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  );
};

export default Calculator;
