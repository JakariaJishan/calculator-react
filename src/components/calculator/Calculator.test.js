import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Calculator from './Calculator';


describe('Calculator page testing', ()=>{
    it('snapshot matching', ()=>{
        const tree = renderer.create(<BrowserRouter><Calculator/></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot()
    })
    
    test('Document testing', ()=>{
        render(<Calculator/>)
        const calcUi = screen.getByText('AC');
        expect(calcUi).toBeInTheDocument()
    })
})
