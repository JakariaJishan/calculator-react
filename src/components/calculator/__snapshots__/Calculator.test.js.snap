// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator page testing snapshot matching 1`] = `
<div
  className="calculator"
>
  <h2
    className="calc-heading"
  >
    Lets do some Math...
  </h2>
  <div
    className="wrapper"
  >
    <div
      className="output"
    >
      0
    </div>
    <div
      className="keys"
    >
      <div
        aria-hidden="true"
        className="ac"
        name="AC"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        AC
      </div>
      <div
        aria-hidden="true"
        className="plus-minus"
        name="+/-"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        +/-
      </div>
      <div
        aria-hidden="true"
        className="percent"
        name="%"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        %
      </div>
      <div
        aria-hidden="true"
        className="division"
        name="รท"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        รท
      </div>
      <div
        aria-hidden="true"
        className="seven"
        name="7"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        7
      </div>
      <div
        aria-hidden="true"
        className="eight"
        name="8"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        8
      </div>
      <div
        aria-hidden="true"
        className="nine"
        name="9"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        9
      </div>
      <div
        aria-hidden="true"
        className="multiply"
        name="x"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        x
      </div>
      <div
        aria-hidden="true"
        className="four"
        name="4"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        4
      </div>
      <div
        aria-hidden="true"
        className="five"
        name="5"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        5
      </div>
      <div
        aria-hidden="true"
        className="six"
        name="6"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        6
      </div>
      <div
        aria-hidden="true"
        className="minus"
        name="-"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        -
      </div>
      <div
        aria-hidden="true"
        className="one"
        name="1"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        1
      </div>
      <div
        aria-hidden="true"
        className="two"
        name="2"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        2
      </div>
      <div
        aria-hidden="true"
        className="three"
        name="3"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        3
      </div>
      <div
        aria-hidden="true"
        className="plus"
        name="+"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        +
      </div>
      <div
        aria-hidden="true"
        className="zero"
        name="0"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        0
      </div>
      <div
        aria-hidden="true"
        className="dot"
        name="."
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        .
      </div>
      <div
        aria-hidden="true"
        className="equal"
        name="="
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        =
      </div>
    </div>
  </div>
</div>
`;
